### PR TITLE
feat(sops): add 3rd party fallback completion loader

### PR DESCRIPTION
**Pull request by scop**
https://github.com/getsops/sops

### CODE REVIEW

File: completions/_sops [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# 3rd party completion loader for commands using           -*- shell-script -*-
+# version 1 of the https://cli.urfave.org library.
+#
+# This serves as a fallback in case the completion is not installed otherwise.
+
+# https://github.com/urfave/cli/blob/v1-maint/docs/v1/manual.md#bash-completion
+# https://github.com/urfave/cli/blob/v1-maint/autocomplete/bash_autocomplete
+_comp_cmd__urfave_cli_v1()
```
Comment (scop) [inline completions/_sops:8]: Not sure if this is according to the naming guideline, but at least it is "in spirit".

File: completions/_sops [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# 3rd party completion loader for commands using           -*- shell-script -*-
+# version 1 of the https://cli.urfave.org library.
```
Comment (scop) [inline completions/_sops:2]: Not sure if there are substantial differences at the moment in the v1 vs v2 completions at the moment, but I initially looked into using the upstream main branch one as a template for `sops`  and didn't get it to work immediately, so thought we might want to add one for each version. That'll mean some more maintenance when tools upgrade their urfave/cli.

- v1: https://github.com/urfave/cli/blob/v1-maint/autocomplete/bash_autocomplete
- v2: https://github.com/urfave/cli/blob/v2-maint/autocomplete/bash_autocomplete
- to-be v3: https://github.com/urfave/cli/blob/main/autocomplete/bash_autocomplete

File: completions/_sops [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    local compcmd=("${words[@]:0:cword}")
+    if [[ $cur == -* ]]; then
+        compcmd+=("$cur")
+    fi
+    compcmd+=(--generate-bash-completion)
+
+    _comp_compgen_split -- "$("${compcmd[@]}" 2>/dev/null)"
+} &&
+    complete -o bashdefault -o default -o nospace \
+        -F _comp_cmd__urfave_cli_v1 "$1"
```
Comment (scop) [inline completions/_sops:22]: Followed upstream in the use of various `-o`. Notably `nospace` does not seem to work that well at least with `sops`, but I don't know if we want to "fix" it and deviate from upstream on this.

**Comment by scop** [comment]
Added upstream FYI discussion at https://github.com/urfave/cli/discussions/1922

**Review by akinomyoga — APPROVED** [review]
I'm not familiar with the upstream framework, but as they seem to have left 👍 , I think this should be fine.

File: completions/_sops (hunk 1 above)
Comment (akinomyoga) [reply completions/_sops:8]: I think this is OK.

**Pull request merged**